### DISCUSSION TITLE
2023年Famland有个新气象

### DISCUSSION THREAD

**Discussion by otto-001** [famland/website#180] [Ideas]
# 梳理了一下famland的现状，如此破局，2023定有新气象

> ## 愿景：
>
> ## 最终的目标打造 Famland 成为一个永动机，永不会死亡，没有中心服务的制约，合约机制存在于链上，靠用户自然驱动成长的一个元宇宙生态。

为了实现这个目标，就目前Farmland的状况，我认为还有两个方面的问题需探讨：

1. 分配问题：

   - 如何根据投资者的贡献完成在流速上的一次分配
   - 制定合理的APY机制完成二次分配。

2. 实现没有中心服务器的技术难点。

   - 前端代码去中心化。

   - 系统周期结算问题。

   - 农场和农田的数据计算和显示问题。

     
 ##  分配问题的思考

目前各种机制对于原始投资人并并不友好，老玩家对系统的信心表示怀疑，尤其是投资原始田的玩家。

根本问题是，我投资了树，我投资了原始田，我为什么还不如新做市的玩家共享值提升的快？

我所在的农场分发池，还没有新做市玩家分发的多，那谁还买树？谁还投资原始田？

不是现在农场分发机制的问题，而是，这些投资树，投资原始田的作用，对农场共享值的提升没有任何作用，反而会影响自己田实际提升共享值的空间。

综上所述，在流速的一次分配上，出现了问题。

#### 第一步：让有原始田的famalnd，所在农场能获得更高的流速分配空间。

> 操作思路：
>
> 改造原始田成为道具NFT。为所在基本田提供共享值加成。当初1000U成本价的原始田，就相当于价值1000U的共享值加成道具NFT。玩家把它加载到那块基本田，那块基本田就获得1000（具体数值可社区讨论）的共享值。
>
> 让技术改造并开放共享值加成NFT（原始田）购买入口，
>
> 1000U每个，其中500U进入多签的开发基金账户，500U直接由稳定池账户配对做市到底池。
>
> 这样做的好处至少有两个：
>
> 1. 就是在FamlandNFT上做减法，把野田也改造成仓库NFT，减少概念，一个Farmland就是一块土地,就是基本田，去掉原始田，野田的概念，降低新玩家学习成本。
> 2. 老玩家的原始田可以在流速分配上占据优势，获得原始投资人的应得的起码尊重。有共享值的加成，各个农场都会欢迎原始田入场。

#### 第二步：按照 讨论区#179方案讨论
[关于APY显示优化的初步思路（待讨论） #179](https://github.com/famland/website/discussions/179)

## 实现没有中心服务器的技术难点

- 首先值得庆祝和肯定的是咱们的前端代码已经通过ENS+IPFS的方案实现去中心化了。
[去中心化的游戏入口web3版本：https://web3.famland.world](https://web3.famland.world/#/)
 
- 目前系统很多数据bug都来自于系统的周期性结算。要如何通过机制去掉周期性结算。从根本上解决bug来源。

> 初步的设想如下：
>
> 目前是按照24小时为一个纪元周期，进行系统结算。现在咱们去掉时间概念。
>
> 不必守着24小时为一个周期， 可以用R的比例来计算达成共识，触发结算，完成纪元周期交替。比如现在有100个参与耕种的账户，100个需要R的， 当达到61.8%的R后，就触发结算，进入下一个周期。
>
> 或者：
>
> 比如A有5块田   B有30块田  加入系统一共有50块田，如果A和B都R过了，那么就是35/50 达到可以执行的条件，执行就可以结算周期。
>
> 或者：
>
> 引入共享值，当共享值达到200或者500的才算有效R，当有效R比例达到61.8%触发结算。
>
> 只看账号或者田的话，有被资本和黑客攻击的风险，加入共享值，就从机制上杜绝了资本和黑客的风险。

- 农场和农田的数据计算和显示问题

  目前农场和农田数据，如肥力值、共享值、apy等等这些数据的展示都是基于中心服务器定期从链上表采集数据后，提供接口给前端排序显示。

  把问题说严重一点，如果中心服务器没有钱续费了，这些数据就会报错或者打不开，显示不出来。玩家就没法正常玩耍了。要解决这个问题，还需要从技术和机制上多方面开脑洞。

> 初步思路：
>
> 在每个数据展示的页面加一个加载数据的按钮，玩家需要查看最新的数据，就自己点加载数据指令给合约，然后计算最新的数据并展示。全过程链上操作，用户承担调用合约的资源。
>
> 
>
> 比如你想把查某个结果，需要按照什么方式排序，那么你执行一下，支付相应的cpu计算资源和net资源，以及Ram资源，这个计算出来的表单就暂用你自己钱包的Ram里，然后你就可以得到了。不然通过合约查询的信息，需要有计算资源，需要带宽资源，以及一个存放的地儿啊，你自己不提供谁来提供呢？
>
> 
>
> 这样每个玩家都是系统的节点，就不需要中心化服务器来服务了。

  

  至此，解决分配问题，彻底去中心化后，就可以把合约所有接口做标准化，公布api文档。开放给其他开发者使用。Famalnd从现实角度来说，具备了不死之身，任何人都不用担心项目方跑路，不会担心自己投入打水漂。

  

  希望大家积极讨论，把项目重新做一次游戏化包装，更新玩法和攻略。

  2023年Famland有个新气象

**Comment by otto-001** [comment]
基于原始田改造成共享值NFT后，需要取消取消原始田收益，同时免除原始田做市收益的效能，只为Famland增加共享值加成效应。
这样做，减少新玩家学习成本，避免跟基本田收益提升共享值之间造成干扰。
减法改造后，玩家只需要理解基本田耕种，收益后获得共享值提升。

**Comment by otto-001** [comment]
原始田和野田做成NFT或扩展仓的思路，可以总体做个架构的梳理：
一、以基本田为根本，作为其他即插即用NFT的主体，并对外开放基本田的API接口和标准。
NFT有四种类型：
1、八宝树；有固定的初始肥力加成系数，两倍；
2、原始田：共享值功能道具
3、外接的NFT：后期投票决定初始加成系数。
4、挂件NFT：初始加成系数为0。
这四种系数都是可以后续通过消耗YL进行打造升级的，类似于波场八宝树的基因升级，这里是解决YL的消耗场景；挂件，是纯为了那些没钱投资的玩家设置的，他们可以消耗时间、消耗YL来打造，打造之后可以以市场上出售，这里是解决引流的问题，让更多玩家无成本或低成本先接触，再投资。
二、结算周期
三、APY
四、游戏包装

**Comment by otto-001** [comment]
还有一个重要问题： 
目前价格很低，无常滑点出发空间很小， 玩家想买入YL做市，也困难。只要是由于稳定池用于支付无常滑点的20%量太少。
以前693.6YL是锚定1U来计算的，那么发行的额度，建议调整为693.5U等值的YL数量。还是按照世界银行贫困线来计算，而不是693.5YL固定数值，在yl的价值没有被认同前，我们还是必须锚定U的价值，流速会根据价格来控制，发行更多出来是进稳定池，让更多用户可以买到，不至于稳定池出现yl余额不足的情况，但是现在yl价格越低，分发到稳定池的量会越来越少。

所以稳定池20%的量要越大越好，这样买进就不会有障碍，不然yl价格越来越低，进入稳定池流入的量还是那么点，远远不够市场消耗。

**Reply by famland16816** [reply]
这个YL的增发，根据第六期的优化方案，原始田可以提升共享值，那么我建议这个增发，
按每块田的投入价值以及当前基准价增发。比如一块田基本田投入是价值500u，原始田投入价值1000u，当前基准价为0.5。那么增发量等于500+1000=1500÷0.5=3000
